pkg/scheduler/framework/plugins/clusteraffinity/types_azure_test.go

```diff
@@ -0,0 +1,257 @@
+				labels.AzureSubscriptionIDLabel: "1234-5678-9012",
+			},
+		},
+	}
+
+	tests := []struct {
+		name          string
+		cluster       *clusterv1beta1.MemberCluster
+		selector      placementv1beta1.PropertySelectorRequirement
+		wantHandled   bool
+		wantAvailable bool
+		wantErr       bool
+	}{
+		{
+			name:    "Azure SKU capacity property not handled",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/count",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"5"},
+			},
+			wantHandled: false,
+		},
+		{
+			name:    "Azure SKU capacity property handled and available",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled:   true,
+			wantAvailable: true,
+		},
+		{
+			name:    "Azure SKU capacity property handled but not available",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/NonExistentSKU/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled:   true,
+			wantAvailable: false,
+		},
+		{
+			name:    "Azure SKU capacity property with invalid operator",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorEqualTo,
+				Values:   []string{"2"},
+			},
+			wantHandled: true,
+			wantErr:     true,
+		},
+		{
+			name:    "Azure SKU capacity property with non-integer value",
+			cluster: cluster,
+			selector: placementv1beta1.PropertySelectorRequirement{
+				Name:     "kubernetes.azure.com/vm-sizes/Standard_D2s_v3/capacity",
+				Operator: placementv1beta1.PropertySelectorGreaterThanOrEqualTo,
+				Values:   []string{"two"},
+			},
+			wantHandled: true,
+			wantErr:     true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create mock server
+			server := createMockAttributeBasedVMSizeRecommenderServer(t, http.StatusOK)
+			defer server.Close()
+
+			client, err := compute.NewAttributeBasedVMSizeRecommenderClient(server.URL, http.DefaultClient)
+			if err != nil {
+				t.Fatalf("failed to create VM size recommender client: %v", err)
+			}
+
+			req := &clusterRequirement{
+				placementv1beta1.ClusterSelectorTerm{},
+				checker.NewPropertyChecker(*client),
+			}
+			handled, available, err := req.MatchPropertiesInPropertyChecker(tt.cluster, tt.selector)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("MatchPropertiesInPropertyChecker() error = %v, wantErr %v", err, tt.wantErr)
+				return
```

Comment:
we cannot return here? as i
```suggestion
				t.Fatalf("MatchPropertiesInPropertyChecker() error = %v, wantErr %v", err, tt.wantErr)
```
Errorf + return = fatalf